src/video_core/query_cache/bank_base.h

```diff
@@ -0,0 +1,106 @@
+// SPDX-FileCopyrightText: Copyright 2023 yuzu Emulator Project
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#pragma once
+
+#include <atomic>
+#include <deque>
+#include <utility>
+
+
+#include "common/common_types.h"
+
+namespace VideoCommon {
+
+class BankBase {
+protected:
+    const size_t base_bank_size;
+    size_t bank_size;
+    std::atomic<size_t> references;
+    size_t current_slot;
+
+public:
+    BankBase(size_t bank_size_)
+        : base_bank_size{bank_size_}, bank_size(bank_size_), references(0), current_slot(0) {}
```

Comment:
```suggestion
        : base_bank_size{bank_size_}, bank_size(bank_size_) {}
```

These members can be initialized directly in the class